webapp/app/plugins/controller.js

```diff
@@ -126,11 +126,17 @@ treeherder.controller('PluginCtrl', [
                         // easier.
                         $scope.job_details = jobInfoArtifact.reduce(function(result, artifact) {
                           if (artifact.blob && Array.isArray(artifact.blob.job_details)) {
-                            result = result.concat(artifact.blob.job_details);
+                              result = result.concat(artifact.blob.job_details);
+                          }
+                          if ($scope.artifacts.buildapi) {
+                              $scope.artifacts.buildapi.blob.title = "Buildername";
+                              $scope.artifacts.buildapi.blob.value = $scope.artifacts.buildapi.blob.buildername;
+                              result = result.concat($scope.artifacts.buildapi.blob);
```

Comment:
This is the part that needs scrutiny, conceptually. We move the buildapi blob here, and give it suitable `title` and `value` properties so it is consistent with the rest of the array objects, and so it works with the angular template which constructs the Job details content. Doing this also gives us an accurate 'count' of Job details entries in the tab.